sysdump/sysdump.go

```diff
@@ -869,6 +870,33 @@ func (c *Collector) Run() error {
 				return nil
 			},
 		},
+		{
+			CreatesSubtasks: true,
+			Description:     "Collection kvstore data",
+			Quick:           true,
+			Task: func(ctx context.Context) error {
+				cm, err := c.Client.GetConfigMap(ctx, c.Options.CiliumNamespace, ciliumConfigMapName, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to get cilium-config ConfigMap: %w", err)
+				}
+				// Check if kvstore is enabled, if not skip this dump.
+				if v, ok := cm.Data["kvstore"]; !ok || v != "etcd" {
+					c.logDebug("KVStore not enabled, skipping kvstore dump")
+					return nil
+				}
+				ps, err := c.Client.ListPods(ctx, c.Options.CiliumNamespace, metav1.ListOptions{
+					LabelSelector: c.Options.CiliumLabelSelector,
+				})
+				if err != nil {
```

Comment:
Suggestion: loop over pods, pick one that is scheduled and (ideally) has all containers running.